src/key.rs

```diff
@@ -26,11 +27,36 @@ use std::str::FromStr;
 ///
 /// To parse a key use `FromStr` trait implementation: `"string".parse::<Key>()`.
 #[derive(Debug, Eq, PartialEq, PartialOrd, Ord, Hash, Clone)]
-pub struct Key {
+pub struct SimpleKey {
+    // Repr.raw_value have things like single quotes.
+    pub(crate) decor: Decor,
+    pub(crate) raw: InternalString,
     key: InternalString,
+}
+
+// Generally, a key is made up of a list of simple-key's separated by dots.
+#[derive(Debug, Eq, PartialEq, PartialOrd, Ord, Hash, Clone)]
+pub struct Key {
     raw: InternalString,
+    pub(crate) parts: Vec<SimpleKey>,
```

Comment:
do we need to store `raw` representation or is it always constructible from the parts?